### PR TITLE
Update document level access control docs

### PR DESCRIPTION
**Pull request by mattgotteiner**
## Purpose

* Make documentation for adding document level access control more clear that there's 2 supported ways directly in the sample.
* Update diagram to indicate the 2 ways clearly
* Reformat / update the .md documentation

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: docs/login_and_acl.md [hunk 1]
```diff
@@ -129,7 +128,7 @@ If you are running setup for the first time, ensure you have run `azd env set AZ
...
 
-In both the chat and ask a question modes, under **Developer settings** optional **Use oid security filter** and **Use groups security filter** checkboxes will appear. The oid (User ID) filter maps to the `oids` field in the search index and the groups (Group ID) filter maps to the `groups` field in the search index. Use the optional scripts included in the sample to manage values for these fields.
+In both the chat and ask a question modes, under **Developer settings** optional **Use oid security filter** and **Use groups security filter** checkboxes will appear. The oid (User ID) filter maps to the `oids` field in the search index and the groups (Group ID) filter maps to the `groups` field in the search index. Once [data with access control information](#add-access-control-data) is present, the filters will restrict results for the logged in user correctly. If `AZURE_ENFORCE_ACCESS_CONTROL` has been set, then access control information is required to get any results at all.
```
Comment (pamelafox) [inline docs/login_and_acl.md:131]: "access control information" isn't clear to me. Do you mean that those boxes must be checked? I think we auto-check them, right, and we don't let them disable if ENFORCE_ACCESS_CONTROL is true:

```
  <Checkbox
      className={styles.chatSettingsSeparator}
      checked={useOidSecurityFilter || requireAccessControl}
      label="Use oid security filter"
      disabled={!isLoggedIn(client) || requireAccessControl}
      onChange={onUseOidSecurityFilterChange}
  />
```

Or are you implying something about the index here?

**Comment by mattgotteiner** [comment]
@pamelafox  we are simultaneously checking for presence of locales in 1 check, but another check is complaining they are there and they should be removed
https://github.com/Azure-Samples/azure-search-openai-demo/pull/1511#issuecomment-2045878415

File: docs/login_and_acl.md (hunk 1 above)
Comment (mattgotteiner) [reply docs/login_and_acl.md:131]: the wording is bad here, I can be more explicit. Good callout

File: docs/login_and_acl.md [hunk 2]
```diff
@@ -138,9 +137,34 @@ In both the chat and ask a question modes, under **Developer settings** optional
...
 
-## Optional scripts
+<a href="add-access-control-data" ></a>
 
-Two optional scripts are provided that allow easier setup of sample data with document level access control. Either the [Data Lake Storage Gen2 script](#azure-data-lake-storage-gen2-setup) or the [manual access control management](#manually-managing-document-level-access-control) can be used to setup sample data for the document level access control system.
+## Adding data with document level access control
+
+The sample supports 2 main strategies for adding data with document level access control.
+
+* [Using the Add Documents API](#api-updates). Sample scripts are provided which directly use the [Add Documents API](https://learn.microsoft.com/rest/api/searchservice/documents/?view=rest-searchservice-2023-11-01&tabs=HTTP) to manage access control information on documents in the index.
```
Comment (pamelafox) [inline docs/login_and_acl.md:146]:
```suggestion
* [Using the Add Documents API](#api-updates). Sample scripts are provided which use the Azure AI Search Service [Add Documents API](https://learn.microsoft.com/rest/api/searchservice/documents/?view=rest-searchservice-2023-11-01&tabs=HTTP) to directly manage access control information on _existing documents_ in the index.
```

File: docs/login_and_acl.md [hunk 3]
```diff
@@ -138,9 +137,34 @@ In both the chat and ask a question modes, under **Developer settings** optional
...
+* [Using prepdocs and Azure Data Lake Storage Gen 2](#azure-data-lake-storage-gen2-setup). Sample scripts are provided which set up an [Azure Data Lake Storage Gen 2](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-introduction) account and set the [access control information](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-access-control) on files and folders stored there.
```
Comment (pamelafox) [inline docs/login_and_acl.md:147]:
```suggestion
* [Using prepdocs and Azure Data Lake Storage Gen 2](#azure-data-lake-storage-gen2-setup). Sample scripts are provided which set up an [Azure Data Lake Storage Gen 2](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-introduction) account, set the [access control information](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-access-control) on files and folders stored there, and ingest those documents into the search index with their  access control information.
```

**Review by pamelafox — APPROVED** [review]
I think there's an open issue about the ADLS2 approach also unnecessarily replicating the content in blob storage, but the explanation looks good assuming that bug gets fixed soon.

**Pull request merged**